python/thunder/rdds/fileio/imagesloader.py

```diff
@@ -199,35 +199,56 @@ def fromTif(self, dataPath, ext='tif', startIdx=None, stopIdx=None, recursive=Fa
 
         def multitifReader(idxAndBuf):
             idx, buf = idxAndBuf
+            pageCount = -1
+            values = []
             fbuf = BytesIO(buf)
             multipage = Image.open(fbuf)
-            pageIdx = 0
-            imgArys = []
-            npagesLeft = -1 if nplanes is None else nplanes  # counts number of planes remaining in image if positive
-            values = []
-            while True:
-                try:
-                    multipage.seek(pageIdx)
-                    imgArys.append(conversionFcn(multipage))
-                    pageIdx += 1
-                    npagesLeft -= 1
-                    if npagesLeft == 0:
-                        # we have just finished an image from this file
-                        retAry = dstack(imgArys) if len(imgArys) > 1 else imgArys[0]
-                        values.append(retAry)
-                        # reset counters:
-                        npagesLeft = nplanes
-                        imgArys = []
-                except EOFError:
-                    # past last page in tif
-                    break
-            if imgArys:
-                retAry = dstack(imgArys) if len(imgArys) > 1 else imgArys[0]
-                values.append(retAry)
+            if multipage.mode.startswith('I') and 'S' in multipage.mode:
+                # signed integer tiff file; use tifffile module to read
+                import thunder.rdds.fileio.tifffile as tifffile
+                fbuf.seek(0)  # reset pointer after read done by PIL
+                tfh = tifffile.TiffFile(fbuf)
+                ary = tfh.asarray()  # ary comes back with pages as first dimension, will need to transpose
+                pageCount = ary.shape[0]  
+                if nplanes is not None:
+                    values = [ary[i:(i+nplanes)] for i in xrange(0, ary.shape[0], nplanes)]
+                else:
+                    values = [ary]
+                tfh.close()
+                # transpose Z dimension if any, leave X and Y in same order
+                if ary.ndim == 3:
+                    values = [val.transpose((1, 2, 0)) for val in values]
+                    # squeeze out last dimension if singleton
+                    values = [val.squeeze(-1) if val.shape[-1] == 1 else val for val in values]
```

Comment:
Offline I confirmed with @freeman-lab that "this" in this case is "squeezing out the last dimension if it's a singleton", and yes, that's the behavior that we expect. It's asserted (among possibly other cases) in `test_imagesloader:TestImagesFileLoaders.test_fromMultipleMultiTimepointTifs`, where we assert that the expected shape of a multipage tif loaded with `npages=1` is `(70, 75)`. So I think we're good here. :)